component/azstorage/datalake.go

```diff
@@ -553,7 +553,34 @@ func (dl *Datalake) ReadInBuffer(name string, offset int64, len int64, data []by
 
 // WriteFromFile : Upload local file to file
 func (dl *Datalake) WriteFromFile(name string, metadata map[string]*string, fi *os.File) (err error) {
-	return dl.BlockBlob.WriteFromFile(name, metadata, fi)
+	// File in DataLake may have permissions and ACL set. Just uploading the file will override them.
+	// So, we need to get the existing permissions and ACL and set them back after uploading the file.
+
+	fileClient := dl.Filesystem.NewFileClient(filepath.Join(dl.Config.prefixPath, name))
```

Comment:
Should we do this behind a hidden flag so that there is no impact to existing flow?

Reply:
Current implementation is incorrect so better to do this always. If someone comes back with complain on the perf part we may consider that. Other way is to just validate if its default or not, if the permissions are default and no ACLs are applied then we can skip the setCall but even in that case Get call will be done additionally

Reply:
We already have an "HonourACL" flag, we can reuse this as well. This flag basically means whatever ACL is set show permissions in local file according to that and not by the permissions.

Reply:
If we don't have this behind a flag, it will always fail for users using OAuth with `Blob Data Contributor` role, because `SetAccessControl` will only work with `Blob Data Owner` role.